### ISSUE TITLE
Seal TheoryAttribute

### ISSUE THREAD

**Issue by bradwilson** [xunit/xunit#2366]
See https://github.com/xunit/xunit.analyzers/pull/131#issuecomment-596274414

**Comment by Joe-Dunleavy** [comment]
I'm happy to resolve this issue, granted that some clarification is needed with regard to where the test method for this should go: [#2554](https://github.com/xunit/xunit/discussions/2554)

**Comment by Joe-Dunleavy** [comment]
`CulturedTheoryAttribute` already inherits from `TheoryAttribute`. However, it isn't clear that  `CulturedTheoryAttribute` has any reason to exist given that its functionality is never used.

ArgumentsFormatterTests.Enums.NonFlags &  ArgumentsFormatterTests.Enums.Flags use the `[CulturedTheory]` attribute, but they might as well use `[Theory]` instead, as no arguments are being passed to `CulturedTheory`.

I suggest that we remove `CulturedTheoryAttrubute` unless it is there for a reason?

**Comment by bradwilson** [comment]
I think I've changed my mind.

When I wrote this issue, `TheoryAttribute` had no properties, so inheriting from it made no sense. In the intervening months, I added `DisableDiscoveryEnumeration` to it, and [code inside of `TheoryDiscoverer` that uses that property](https://github.com/xunit/xunit/blob/12026778465c166f94521ea634bc923cf05cb3f1/src/xunit.v3.core/Sdk/Frameworks/TheoryDiscoverer.cs#L135-L137). True, it's looked up by name (because of our reflection abstractions in place to support source-based discovery), so if we sealed `TheoryAttribute`, anybody could just add the property with the same name and it would "just work" if that attribute were discovered via `TheoryAttribute`, but that sort of fails the "pit of success" that we'd get if we just continued to let people derive from `TheoryAttribute`.

I'm not gonna close this yet, but I'm strongly considering it. Just want to mull it over.
